### PR TITLE
fix: Uppercase issues in Navigation Titles in `ConversationTitleView` and `ConversationDetails` - ACC - 292

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we fix uppercase issues for Navigation titles in all screens within `Conversation Details` and in `ConversationTitleView`. In addition to that within `Conversation Details` we fixed the title that appears in the Navigation Bar in Guests Or Services. Now instead of the name of the conversation we show Guests or Services accordingly. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/ConversationTitleView.swift [hunk 1]
```diff
@@ -59,7 +59,7 @@ final class ConversationTitleView: TitleView {
         }
 
         super.configure(icons: attachments,
-                        title: conversation.displayName.localizedUppercase,
+                        title: conversation.displayName.capitalized,
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Conversation/ConversationTitleView.swift:62]: Is there a difference between localized uppercase and capitalized?

Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation/ConversationTitleView.swift:62]: Yeah, @johnxnguyen! localizedUppercase is for all letters in the string. Capitalised is only for the first letter of the word(s) in a string

File: Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift [hunk 2]
```diff
@@ -59,10 +59,6 @@ extension ZMConversation {
             return conversation.teamRemoteIdentifier == selfUser.teamIdentifier
         }
 
-        var title: String {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift:62]: Do you know why we have this property? I think we are using it somewhere on another screen

Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Conversation Options/Conversation+OptionsConfiguration.swift:62]: Yes, I know. We were using it in the navigation titles for the Guests and Services. That's why I removed it now

**Review by KaterinaWire — APPROVED** [review]

**Review by ghuMustafa — APPROVED** [review]
LGTM.

**Pull request merged**